07-Interpretability-of-ML/LRP-L96.ipynb

```diff
@@ -1,5 +1,16 @@
 {
```

Comment:
<p>Are there any references on interpreting NNs for regression using Saliency maps?</p><p>I have not carefully looked through the references here, but reading the abstracts they all seem to be for classification problems. Maybe the extension to regression is trivial, but I (being a ML novice) am finding this notebook slightly hard to follow - put another way I am not sure what to do with the gradient and LRP plots at the end, or what insight is being actually derived there.</p><p> </p><p>I tried to go back and listen to Pierre's talk and that didn't really help me either. Is there a very simple intro level reading that can be attached here?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/m2lines/L96_demo/pull/21/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Reply:
Good point -- I'll look for one. As elaborated in my comment below, the right way to think about gradients is as the local linear approximation of the model (i.e. the linear model that best approximates the NN at a given point, where in our plots we've averaged this model over 200 points). The right way to think about LRP or gradient * input is like that same linear model multiplied by the input, so it's giving the actual contributing evidence of each input to the output.

Reply:
The interpretation might actually be easier/more sensical for regression than classification, actually, because in classification you have to take one additional step and relate the thing being output by the model (the logits / log-odds) back to the thing you actually care about (the probabilities).

Reply:
I added a bit more explanation to the notebook, let me know if this is ok!